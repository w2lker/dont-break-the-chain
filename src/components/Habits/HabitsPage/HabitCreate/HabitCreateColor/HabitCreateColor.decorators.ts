import withStyles from '@material-ui/core/styles/withStyles';
import HabitCreateColor from './HabitCreateColor';
import habitCreateColorStyles from './HabitCreateColor.styles';

const styledComponent = withStyles(habitCreateColorStyles)(HabitCreateColor);

export default styledComponent;
