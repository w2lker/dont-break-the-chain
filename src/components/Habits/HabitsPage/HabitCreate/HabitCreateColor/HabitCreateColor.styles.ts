const habitCreateColorStyles = {
  root: {
    padding: 0,
  },
};

export default habitCreateColorStyles;
