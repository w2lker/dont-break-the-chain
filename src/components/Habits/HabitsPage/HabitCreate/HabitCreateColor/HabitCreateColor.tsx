import React from 'react';

import { WithStyles } from '@material-ui/styles';

import habitCreateColorStyles from './HabitCreateColor.styles';

export interface IHabitCreateColorProps extends WithStyles<typeof habitCreateColorStyles> {}

const HabitCreateColor: React.FC<IHabitCreateColorProps> = (props) => {
  const { classes } = props;
  return (
    <div className={classes.root}>
      HabitCreateColor Component
    </div>
  );
};

export default HabitCreateColor;
