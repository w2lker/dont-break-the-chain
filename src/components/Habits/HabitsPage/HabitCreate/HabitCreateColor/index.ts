import HabitCreateColor from './HabitCreateColor.decorators';

export default HabitCreateColor;
