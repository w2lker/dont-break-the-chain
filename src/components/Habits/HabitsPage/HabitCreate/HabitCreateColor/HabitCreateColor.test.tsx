import React from 'react';

import { mount, shallow } from 'enzyme';
// import cases from 'jest-in-case';

import HabitCreateColor, { IHabitCreateColorProps } from './HabitCreateColor';
import HabitCreateColorDecorated from './HabitCreateColor.decorators';
import habitCreateColorStyles from './HabitCreateColor.styles';

const testSetup = () => {
  const classes = {
    root: 'habitCreateColor-wrapper',
  };
  const classesKeys = Object.keys(classes);
  const emptyProps: IHabitCreateColorProps = {
    classes,
  };
  const sampleProps: IHabitCreateColorProps = {
    classes,
  };
  return { classes, classesKeys, emptyProps, sampleProps };
};

describe('HabitCreateColor component', () => {
  it('match snapshot', () => {
    const { sampleProps } = testSetup();
    const component = shallow(<HabitCreateColor {...sampleProps} />);
    expect(component.debug()).toMatchSnapshot();
  });

  test.todo('sample placeholder');
});

describe('HabitCreateColor styles', () => {
  it('contains principle classes', () => {
    const { classesKeys } = testSetup();
    expect(habitCreateColorStyles).toBeDefined();
    classesKeys.forEach((keyValue) => {
      // @ts-ignore
      expect(habitCreateColorStyles[keyValue]).toBeDefined();
    });
  });
});

describe('HabitCreateColor decorators', () => {
  it('provides styled classes from decorators', () => {
    const { sampleProps, classesKeys } = testSetup();
    const component = mount(<HabitCreateColorDecorated {...sampleProps} />);
    // @ts-ignore
    const assignedClasses = component.find('HabitCreateColor').props().classes;
    classesKeys.forEach(keyValue => {
      expect(assignedClasses[keyValue]).toBeDefined();
    });
  });
});
