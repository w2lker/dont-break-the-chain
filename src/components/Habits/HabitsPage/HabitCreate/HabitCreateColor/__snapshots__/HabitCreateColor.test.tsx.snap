// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HabitCreateColor component match snapshot 1`] = `
"<div className=\\"habitCreateColor-wrapper\\">
  HabitCreateColor Component
</div>"
`;
